src/posix/net.cpp

```diff
@@ -35,13 +35,19 @@ MUTEX_CLASS CurlHttpIO::curlMutex(false);
 #if defined(USE_OPENSSL) && !defined(OPENSSL_IS_BORINGSSL)
 
 MUTEX_CLASS **CurlHttpIO::sslMutexes = NULL;
+static MUTEX_CLASS lock_init_mutex;
```

Comment:
The constructor needs a parameter to initialize the mutex. Otherwise, most implementations of `MUTEX_CLASS` expect a call to `init()` and crash in the first `lock()` if it's not called. The only exception is the implementation for Win32.
Changing this line by `static MUTEX_CLASS lock_init_mutex(false);` (or `true` but the mutex doesn't need to be recursive in this case) the fix should work for all platforms.

Reply:
thanks for catching that - I've added the parameter, cheers